frontend/src/components/pages/CampOverview/EditCamperModal/EditCamperFormResponseSection.tsx

```
@@ -0,0 +1,175 @@
+        <FormLabel>
+          <Text fontSize="md" as="b">
+            {question}
+          </Text>
+        </FormLabel>
+        <CheckboxGroup
+          defaultValue={answeredOptions}
+          onChange={(newAnswers) => {
+            const newAnswer = newAnswers.join(",");
+            updateFormResponse(question, newAnswer);
+          }}
+        >
+          <Stack direction="column">
+            {options.map((option) => (
+              <Checkbox key={`${questionId}_${option}`} value={option}>
+                {option}
+              </Checkbox>
+            ))}
+          </Stack>
+        </CheckboxGroup>
+      </FormControl>
+    );
+  };
+
+  return (
+    <VStack align="start">
+      {formQuestions.map((formQuestion) => {
+        const { question, type } = formQuestion;
+        const camperAnswerIndex = Object.keys(camperResponses).indexOf(
+          question,
+        );
+        const camperAnswer =
+          camperAnswerIndex !== -1
+            ? Object.values(camperResponses)[camperAnswerIndex]
+            : "";
+
+        switch (type) {
+          case "MultipleChoice":
+            return formQuestion.options
+              ? renderMCQ(
+                  question,
+                  camperAnswer,
+                  formQuestion.options,
+                  formQuestion.id,
+                  formQuestion.required,
+                )
+              : null;
+          case "Multiselect":
+            return formQuestion.options
+              ? renderMultiSelect(
+                  question,
+                  camperAnswer,
+                  formQuestion.options,
+                  formQuestion.id,
+                  formQuestion.required,
+                )
+              : null;
+          case "Text":
+            return renderText(
+              question,
+              camperAnswer,
+              formQuestion.id,
+              formQuestion.required,
+            );
+          default:
+            return null;
```

If this is exhaustive, we should probably do some sort of error here?

---

Yeah so this should never really be the case since the value is retrieved from the backend. What kinda error handling were you thinking?

---

Hmmm I'm not sure what the patterns are here: logging, alert, throw error (don't remember if it's swallowed once deployed?)?